testsuite/MDAnalysisTests/coordinates/base.py

```diff
@@ -338,7 +346,46 @@ def test_stop_iter(self, reader):
         for ts in reader[:-1]:
             pass
         assert_equal(reader.frame, 0)
-
+        
+    def test_transformations_iter(self, ref, transformed):
+        # one iteration
+        for i, ts in enumerate(transformed):
+            idealcoords = ref.iter_ts(i).positions + (1,1,1)
+            assert_array_almost_equal(ts.positions, idealcoords, decimal=ref.prec)
+    
+    def test_transformations_2iter(self, ref, transformed):
+        # two iterations
+        idealcoords=[]
+        for i, ts in enumerate(transformed):
+            idealcoords.append(ref.iter_ts(i).positions + (1,1,1))
+            assert_array_almost_equal(ts.positions, idealcoords[i], decimal=ref.prec)
+            
+        for i, ts in enumerate(transformed):
+            assert_almost_equal(ts.positions, idealcoords[i], decimal=ref.prec)
+            
+    def test_transformations_slice(self, ref, transformed):
+        for i,ts in enumerate(transformed[2:3:1]):
+            idealcoords = ref.iter_ts(ts.frame).positions + (1,1,1)
+            assert_array_almost_equal(ts.positions, idealcoords, decimal = ref.prec)
+            
+        
+    def test_transformations_switch_frame(self, ref, transformed):
+        # switch to a different frame
+        first_ideal = ref.iter_ts(0).positions + (1,1,1)
+        if len(transformed)>1:
+            assert_array_almost_equal(transformed[0].positions, first_ideal, decimal = ref.prec)
```

Comment:
You could comment on what you are testing for each assert. I understand it now because I have the context, but when the test will fail the context will be long gone. A comment that states the question you are answering for each assert in that function would help.